NPE in TOR when students' credit has no grade

NPE in TOR when students' credit has no grade.

Caused by: java.lang.NullPointerException
    at fi.otavanopisto.muikku.plugins.assessmentrequest.AssessmentRequestController.getWorkspaceAssessmentState(AssessmentRequestController.java:139)

--- a/muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/assessmentrequest/AssessmentRequestController.java
+++ b/muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/assessmentrequest/AssessmentRequestController.java
@@ -135,6 +135,10 @@ public class AssessmentRequestController {
     if (latestAssessment != null) {
       gradingScale = gradingController.findGradingScale(latestAssessment.getGradingScaleIdentifier());
       grade = gradingController.findGradingScaleItem(gradingScale, latestAssessment.getGradeIdentifier());
+      
+      if (gradingScale == null || grade == null) {
+        latestAssessment = null;
+      }
     }
     if (latestAssessment != null && (latestRequest == null || latestRequest.getDate().before(latestAssessment.getDate()))) {
 

